ccvm_simulators/solvers/langevin_solver.py

```diff
@@ -284,6 +284,69 @@ def _fpga_machine_energy_callable(matching_df: DataFrame, problem_size: int):
 
         return _fpga_machine_energy_callable
 
+    def _fpga_machine_time(self, machine_parameters: dict = None):
+        """The wrapper function of calculating the average time spent by the
+            solver on a single instance, as if the solving process was to be performed on
+            an FPGA machine.
+
+        Args:
+            machine_parameters (dict, optional): Parameters of the FPGA machine.
+                Defaults to None.
+
+        Raises:
+            ValueError: when the given machine parameters are not valid.
+            ValueError: when the given dataframe does not contain the required columns.
+
+        Returns:
+            Callable: A callable function that takes in a dataframe and problem size and
+                returns the average time spent by the solver on a single instance.
+        """
+
+        if machine_parameters is None:
+            machine_parameters = self._default_fpga_machine_parameters
+        else:
+            self._validate_fpga_machine_parameters(machine_parameters)
+
+        def _fpga_machine_time_callable(dataframe: DataFrame, problem_size: int):
+            """Calculate the average time spent by the solver on a single instance,
+                simulating on an FPGA machine.
+
+            Args:
+                dataframe (DataFrame): The necessary data to calculate the average
+                    time.
+                problem_size (int): The size of the problem.
+
+            Raises:
+                ValueError: when the given dataframe does not contain the required
+                    columns.
+
+            Returns:
+                float: The average time spent by the solver on a single instance.
+            """
+            try:
+                postprocessing_time = np.mean(dataframe["pp_time"].values)
+            except KeyError as e:
+                missing_column = e.args[0]
+                raise ValueError(
+                    f"The given dataframe is missing the {missing_column} column. Required columns are: ['pp_time']."
```

Comment:
I'm not entirely clear on this error message. If the required column is `pp_time`, wouldn't the missing column in this case be `pp_time`?"

Reply:
That's a great point. I was going for consistency with the other solvers, but for the user that doesn't make a lot of sense. I'll update it!